sys/kern/subr_vmem.c

```diff
@@ -960,6 +1016,7 @@ vmem_fit(const bt_t *bt, vmem_size_t size, vmem_size_t align,
 	/*
 	 * XXX assumption: vmem_addr_t and vmem_size_t are
 	 * unsigned integer of the same size.
+	 * CHERI breaks this assumption.
```

Comment:
What are the implications of that? If that's been addressed then the XXX should go? Well, unless it's only fixed (a) when it's purecap and (b) for the specific differences that apply to purecap?

Reply:
The XXX comes from upstream in one of the early vmem commits. I believe it is there because of the CROSS_P and alignment macros, that may have had weird results if one of the types was uint32_t? I'm not sure it is relevant anymore, surely is fixed for purecap as the size is different but does not affect the actual vaddr size.